### PR TITLE
[CIT-646] Add child device support doc

### PR DESCRIPTION
**Pull request by rina23q**
Signed-off-by: Rina Fujino <18257209+rina23q@users.noreply.github.com>

## Proposed changes

Add our new child device support user documents. Also related to #508 #541 

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by itsyitsy — APPROVED** [review]

File: docs/src/architecture/mapper.md [hunk 1]
```diff
@@ -67,6 +67,48 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
...
+The incoming topic is different from the parent device.
+For child devices, use `tedge/measurements/<child-id>` topic. (`<child-id>` is your desired child device ID.)
+
+The mapper works in the following steps.
+
+1. When the mapper receives a Thin Edge JSON message on the `tedge/measurements/<child-id>`,
+   the mapper publishes the SmartREST `101` message to create a child device under the `thin-edge.io` parent device.
+   The name of the child device is the given child ID, and the type is `thin-edge.io-child`.
+2. Publish corresponded Cumulocity JSON measurements messages over MQTT.
+3. If the given child ID by the topic name is already known by the mapper,
```
Comment (albinsuresh) [inline docs/src/architecture/mapper.md:82]: That's an implementation detail and I'm not sure if this level of detail is required in the userdoc. For the user, the relevant thing is that "the child device is created on receipt of the very first measurement for that child device".

File: docs/src/architecture/mapper.md [hunk 2]
```diff
@@ -67,6 +67,48 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
 After the mapper publishes a message on the topic `c8y/measurement/measurements/create`,
 the message will be transferred to the topic `measurement/measurements/create` by [the MQTT bridge](../references/bridged-topics.md).
 
+### For child devices
+
+The Cumulocity mapper supports sending measurements to child devices under the `thin-edge.io` parent device.
```
Comment (didier-wenzek) [inline docs/src/architecture/mapper.md:72]: This sounds a bit incorrect / incomplete. I would say:

```
The Cumulocity mapper collects measurements not only from the main device but also from child devices.
These measurements are collected under the `tedge/measurements/<child-id>` topics
and forwarded to Cumulocity to corresponding child devices created under the `thin-edge.io` parent device.
```

File: docs/src/architecture/mapper.md [hunk 3]
```diff
@@ -67,6 +67,48 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
...
+The incoming topic is different from the parent device.
+For child devices, use `tedge/measurements/<child-id>` topic. (`<child-id>` is your desired child device ID.)
+
+The mapper works in the following steps.
+
+1. When the mapper receives a Thin Edge JSON message on the `tedge/measurements/<child-id>`,
```
Comment (didier-wenzek) [inline docs/src/architecture/mapper.md:78]:
```suggestion
1. When the mapper receives a Thin Edge JSON message on the `tedge/measurements/<child-id>` topic,
```

File: docs/src/architecture/mapper.md [hunk 4]
```diff
@@ -67,6 +67,48 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
...
+   the mapper publishes the SmartREST `101` message to create a child device under the `thin-edge.io` parent device.
+   The name of the child device is the given child ID, and the type is `thin-edge.io-child`.
```
Comment (didier-wenzek) [inline docs/src/architecture/mapper.md:80]:
```suggestion
   The child device is named after the `<child-id>` topic name, and the type is `thin-edge.io-child`.
```

File: docs/src/architecture/mapper.md (hunk 1 above)
Comment (didier-wenzek) [reply docs/src/architecture/mapper.md:82]: Agree with @albinsuresh . The user doc is not a specification. The proposed sentence is perfect.

File: docs/src/architecture/mapper.md [hunk 5]
```diff
@@ -67,6 +67,48 @@ as both a fragment type and a fragment series in [Cumulocity's measurements](htt
...
+    "temperature":{
+      "value":23
+    }
+  }
+}
+```
+
 ## Azure IoT Hub mapper
 
 The Azure IoT Hub mapper takes messages formatted in the [Thin Edge JSON](thin-edge-json.md) as input.
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply docs/src/architecture/mapper.md:114]: Yes adding a single sentence to say that "child device measurements are not supported yet on Azure" would be good.

**Review by didier-wenzek — COMMENTED** [review]
The content is correct and complete, but I think the writing can be improved.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**